dom0/sd-clean-all.sls

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# vim: set syntax=yaml ts=2 sw=2 sts=2 et :
+
+set-fedora-as-default-dispvm:
+  cmd.run:
+    - name: qubes-prefs default_dispvm fedora-30-dvm
+
+remove-dom0-sdw-config-files:
+  file.absent:
+    - names:
+      - /opt/securedrop
+      - /etc/yum.repos.d/securedrop-workstation-dom0.repo
+      - /usr/bin/securedrop-update
+      - /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test
+      - /etc/cron.daily/securedrop-update-cron
+      - /usr/share/securedrop/icons
+
+sd-cleanup-sys-firewall:
+  cmd.run:
+    - names:
+      - qvm-run sys-firewall 'sudo rm -f /rw/config/RPM-GPG-KEY-securedrop-workstation-test'
+      - qvm-run sys-firewall 'sudo rm -f /rw/config/sd-copy-rpm-repo-pubkey.sh'
+      - qvm-run sys-firewall 'sudo rm -f /etc/pki/rpm-gpg/RPM-GPG-KEY-securedrop-workstation-test'
+      - qvm-run sys-firewall 'sudo perl -pi -E "s#^/rw/config/sd-copy-rpm-repo-pubkey.sh##" /rw/config/rc.local'
+
+sd-cleanup-rpc-mgmt-policy:
+  file.replace:
+    - names:
+      - /etc/qubes-rpc/policy/qubes.VMShell
+      - /etc/qubes-rpc/policy/qubes.VMRootShell
+    - repl: ''
+    - pattern: '^disp-mgmt-sd-\w+\s+sd-\w+\s+allow,user=root'
+
+{% set sdw_customized_rpc_files = salt['cmd.shell']('grep -rl "BEGIN securedrop-workstation" /etc/qubes-rpc/ | cat').splitlines() %}
```

Comment:
Awesome! There's one edge case that we could. When vim or other editor leaves a swap file (like it did on my system), the file is viewed as a binary and the `make clean` action will fail. This is obviously an edge case (and deleting that file resolved), see output below. Perhaps grepping for strings that don't start with a dot is sufficient here, but IMO not required for merge.


```
----------
          ID: sd-cleanup-rpc-policy-grants
    Function: file.replace
        Name: /etc/qubes-rpc/policy/.qubes.Filecopy.swp
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib/python2.7/site-packages/salt/state.py", line 1837, in call
                  **cdata['kwargs'])
                File "/usr/lib/python2.7/site-packages/salt/loader.py", line 1794, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib/python2.7/site-packages/salt/states/file.py", line 3934, in replace
                  ignore_if_missing=ignore_if_missing)
                File "/usr/lib/python2.7/site-packages/salt/modules/file.py", line 2035, in replace
                  .format(path)
              SaltInvocationError: Cannot perform string replacements on a binary file: /etc/qubes-rpc/policy/.qubes.Filecopy.swp
     Started: 16:16:15.438709
    Duration: 5.279 ms
     Changes:   

Summary for local
-------------
Succeeded: 28 (changed=26)
Failed:     1
```

Reply:
Hmm, great catch. We can just pass `-I` to grep so it doesn't report binary files. That should handle the edge case you describe.